packages/synapse-react-client/src/components/CitationPopover/CitationPopover.test.tsx

```diff
@@ -0,0 +1,194 @@
+import { act, cleanup, render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { UseQueryResult } from '@tanstack/react-query'
+import CitationPopover from './CitationPopover'
+import { useCitation } from './useCitation'
+
+jest.mock('./useCitation', () => ({
+  useCitation: jest.fn(),
+}))
+
+const mockUseCitation = useCitation as jest.MockedFunction<typeof useCitation>
+
+const openPopover = async () => {
+  const button = screen.getByTestId('CiteAsButton')
+  await act(async () => {
+    await userEvent.click(button)
+  })
+}
+
+const data =
+  '@misc{test2025,\n  title = {Some BibTeX Entry},\n  year = {2025}\n}'
+
+const mockProps = {
+  doi: 'https://doi.org/10.1234/abcd1234',
+  title: 'Some Article',
+  boilerplateText: 'Some boilerplate text',
+}
+
+describe('CitationPopover tests', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+    mockUseCitation.mockReturnValue({
+      isLoading: false,
+      data: data,
+      error: null,
+      isError: false,
+      isSuccess: true,
+    } as UseQueryResult<string, Error>)
+  })
+  afterEach(() => {
+    cleanup()
+  })
+
+  it('renders button', () => {
+    render(<CitationPopover {...mockProps} />)
+    const button = screen.queryByTestId('CiteAsButton')
+    expect(button).toBeInTheDocument()
```

Comment:
Will this work? 2 notes:
1. Prefer `xxxByRole` over `xxxByTestId` -- https://testing-library.com/docs/queries/about#priority
2. `findByXxx` will fail if it doesn't find the element, so it's more succinct than `queryByXxx` followed by `toBeInTheDocument`

```suggestion
    await screen.findByRole('button', { name: 'Cite As' })
```